aiida_project/project/base.py

```diff
@@ -7,7 +7,7 @@
 from pydantic import BaseModel
 
 
-def recursive_mkdir(project_path: Path, structure: dict | list | Path) -> None:
+def recursive_mkdir(project_path: Path, structure: dict | list | Path) -> None:  # type: ignore[type-arg]
```

Comment:
The type ignores here are just for brevity of type annotations ---- otherwise I would have to spell out Any as generic type parameters everywhere...